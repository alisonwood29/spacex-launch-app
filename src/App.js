import './App.css';
import { LaunchPage } from './pages/LaunchPage';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <LaunchPage/>
      </header>
    </div>
  );
}

export default App;
