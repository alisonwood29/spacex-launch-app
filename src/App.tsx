import React from 'react';

import { LaunchPage } from './pages/LaunchPage';

const App: React.FC = () => {
  return (
    <div className="App">
      <header className="App-header">
        <LaunchPage />
      </header>
    </div>
  );
}

export default App;
