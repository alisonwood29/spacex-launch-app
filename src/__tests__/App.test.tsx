import { render } from '@testing-library/react';

import App from '../App';

describe('Test app', () => {
  it('renders', () => {
    const { container } = render(<App />);
    
    expect(container).toBeDefined();
    expect(container).not.toBeNull();
  });

  it('renders a snapshot', () => {
    const { container } = render(<App />);

    expect(container).toMatchSnapshot();
  });
});
