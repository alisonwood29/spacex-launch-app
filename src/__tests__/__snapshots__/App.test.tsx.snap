// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test app renders a snapshot 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <div
        class="launch-page"
      >
        <div
          class="launch-page-header"
        >
          <img
            alt="SpaceX Title Logo"
            class="spacex-logo"
            src="spacex-logo.png"
          />
          <button
            class="refresh-button"
            data-testid="refresh-button"
          >
            Reload data
            <img
              alt="Refresh Icon"
              class="refresh-icon"
              src="refresh.png"
            />
          </button>
        </div>
        <div
          class="launch-page-content"
        >
          <img
            alt="rocket launch logo"
            class="launch-logo"
            src="launch-home.png"
          />
          <div
            class="launch-list-div"
          >
            <div
              class="button-div"
            >
              <button
                class="sort-button"
              >
                <span>
                  Sort 
                  Descending
                </span>
                <img
                  alt="Up and Down Sort Arrows"
                  class="sort-icon"
                  src="sort.png"
                />
              </button>
            </div>
            <div
              class="launch-list"
              data-testid="launch-list"
            />
          </div>
        </div>
      </div>
    </header>
  </div>
</div>
`;
